### PR TITLE
Fixes to git reference selection with promote builds

### PR DESCRIPTION
**Pull request by halkyon**
- Display fields relating to radio option below radio button
 - Only show promote option when deploying to production environments
 - Add build status information when selecting promote option

### CODE REVIEW

File: code/control/GitDispatcher.php [hunk 1]
```diff
@@ -81,17 +81,24 @@ public function update(SS_HTTPRequest $request) {
...
+			$targetEnvironment = DNEnvironment::get()->byId((int) $targetEnvironmentId);
+		}
+
 		$refs = [];
 		$prevDeploys = [];
 
 		$uatEnvironment = $this->project->DNEnvironmentList()->filter('Usage', 'UAT')->first();
 		$uatBuild = $uatEnvironment ? $uatEnvironment->CurrentBuild() : null;
-		if ($uatBuild && $uatBuild->exists()) {
+		if ($uatBuild && $uatBuild->exists() && $targetEnvironment && $targetEnvironment->Usage === 'Production') {
```
Comment (stojg) [inline code/control/GitDispatcher.php]: We should probably change to use the `DNEnvironment::PRODUCTION` constant? I noticed that we also haven't used that in other places :(

Comment (halkyon) [inline code/control/GitDispatcher.php]: Damnit.. yes, you're right :)

File: js/reducers/environment.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
+var _ = require('underscore');
+
+var actions = require('../_actions.js');
+
 module.exports = function environment(state, action) {
 	if (typeof state === 'undefined') {
 		return {
-			name: ""
+			id: "",
```
Comment (stojg) [inline js/reducers/environment.js]: is this an int? should the default be null?

File: js/reducers/git.js [hunk 3]
```diff
@@ -80,7 +80,6 @@ module.exports = function git(state, action) {
 			if (action.list.refs.constructor === Array) {
 				listAsArray = _.assign({}, action.list.refs);
 			}
-			console.log(listAsArray);
```
Comment (stojg) [inline js/reducers/git.js]: oops. sorry.

File: js/reducers/environment.js (hunk 2 above)
Comment (halkyon) [inline js/reducers/environment.js]: yeah, I was rolling with what was already there. I'll make them null.

Comment (halkyon) [inline js/reducers/environment.js]: fixed

File: code/control/GitDispatcher.php (hunk 1 above)
Comment (halkyon) [inline code/control/GitDispatcher.php]: fixed

File: js/reducers/git.js (hunk 3 above)
Comment (halkyon) [inline js/reducers/git.js]: no worries, I was clearing out the easy stuff from lint

**Pull request merged**